src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorDefinitionProvider.ts

```diff
@@ -0,0 +1,39 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorLanguageFeatureBase } from './RazorLanguageFeatureBase';
+
+export class RazorDefinitionProvider
+    extends RazorLanguageFeatureBase
+    implements vscode.DefinitionProvider {
+
+    public async provideDefinition(
+        document: vscode.TextDocument, position: vscode.Position,
+        token: vscode.CancellationToken) {
+
+        const projection = await this.getProjection(document, position, token);
+        if (projection) {
+            const definitions = await vscode.commands.executeCommand<vscode.Definition>(
+                'vscode.executeDefinitionProvider',
+                projection.uri,
+                projection.position);
+            // C# knows about line pragma, if we're getting a direction to a virtual c# document
+            // that means the piece we're trying to navigate to does not have a representation in the
+            // top level file.
+            const result = (definitions as vscode.Location[]).filter(element => {
```

Comment:
```suggestion
            const result = (definitions as vscode.Location[]).filter(definition => {
```

Reply:
Surround this with `if (projection.languageKind == LanguageKind.CSharp)` and then put the above comment inside. No need to filter if the projection isn't C# related. Ditto with the below HTML condition which does a `forEach`